### PR TITLE
feat:APIs for user authentication

### PR DESCRIPTION
**Pull request by JainishaParmar**
added APIs for authentication

### CODE REVIEW

**Review by jaimin45 — APPROVED** [review]

**Review by jaimin45 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by jaimin45 — APPROVED** [review]

File: swagger.yaml [hunk 1]
```diff
@@ -137,30 +142,42 @@ paths:
...
           description: 'Internal server error'
           schema:
             $ref: '#/definitions/500Response'
+
+  /users/{id}:
+    parameters:
+      - name: id
+        in: path
+        required: true
+        description: Tutorial id
```
Comment (kapilbarad) [inline swagger.yaml]: Need to correct description.

File: swagger.yaml [hunk 2]
```diff
@@ -169,66 +186,99 @@ paths:
...
+          required: true
+          schema:
+            $ref: '#/definitions/OauthLogin'
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/OauthToken'
+        '401':
+          description: 'User is unauthorized'
```
Comment (kapilbarad) [inline swagger.yaml]: Just `Unauthorized` would be fine.

File: swagger.yaml [hunk 3]
```diff
@@ -169,66 +186,99 @@ paths:
...
+            $ref: '#/definitions/404Response'
+        '500':
+          description: 'Internal Server Error'
+          schema:
+            $ref: '#/definitions/500Response'
       security:
         - JWT: []
 
-  /users/oauth/reset-password:
+  /oauth/reset-password:
```
Comment (kapilbarad) [inline swagger.yaml:237]: We need 401 response for this API, so we can return Unauthorized when provided invalid expired JWT token.

File: swagger.yaml [hunk 4]
```diff
@@ -169,66 +186,99 @@ paths:
...
+      summary: Login user
+      tags:
+        - Authentication
+      parameters:
+        - name: 'Login'
+          in: 'body'
+          description: 'Enter email and password for login'
+          required: true
+          schema:
+            $ref: '#/definitions/OauthLogin'
```
Comment (kapilbarad) [inline swagger.yaml]: Should be PostOauthTokenBody

**Review by kapilbarad — CHANGES_REQUESTED** [review]

**Review by kapilbarad — APPROVED** [review]

**Comment by JainishaParmar** [comment]
feat:APIs for user Authentication

**Pull request merged**